context/src/repo/mod.rs

```diff
@@ -155,6 +163,53 @@ impl BundleRepo {
         }
         Ok(None)
     }
+
+    #[cfg(feature = "git-access")]
+    fn resolve_repo_head_commit<'a>(
+        git_repo: &'a gix::Repository,
+        current_commit: gix::Commit<'a>,
+        repo_head_branch: String,
+    ) -> gix::Commit<'a> {
+        // for GH actions, grab sha of PR head, not the PR merge commit
+        let re = Regex::new(r"refs\/remotes\/pull\/[0-9]+\/merge").unwrap();
+        if re.is_match(&repo_head_branch) && current_commit.parent_ids().count() == 2 {
+            log::debug!("attempting to resolve PR head commit");
+
+            let branch_to_fetch = repo_head_branch.replace("remotes/", "");
+            match Command::new("git")
```

Comment:
I believe `gix` has a fetch command we can use instead of running a command - https://docs.rs/gix/latest/gix/clone/fetch/index.html (this is probably linking the wrong thing lol)

Reply:
I tried extensively to implement this with `gix` but was unsuccessful every time. I think it's implementation of `fetch` right now is half-baked, or only used internally for it's `clone` module

Reply:
unrelated to this PR: we should probably switch to [`git2`](https://crates.io/crates/git2) if `gix` isn't cutting the mustard